landscape/lib/message.py

```diff
@@ -10,6 +10,12 @@ def got_next_expected(store, next_expected):
     wants next; this will do various things based on what *this* side
     has in its outbound queue store.
 
+    0. The peer expects a sequence number greater than the number of messages
+       we just sent. This means something is not right. We advance the pending
+       offset from the previous payload since the server does not want old or
+       ancient messages, however we also reset it so that current payload is
+       resent. Then we resynchronize by calling RESYNC. See LP: #1917540
+
```

Comment:
That description (0) doesn't seem accurate. You don't advance the pending offset, you just flush the outgoing queue.

Also the "sequence number greater than the number of messages we just sent" part seems unsound. There can be more pending messages than sent and still be fine, since the broker has a maximum payload size. [see exchange](../../blob/master/landscape/client/broker/exchange.py#L687)

Reply:
I agree that your wording about the flushing outgoing queue is more accurate so I am changing it.

Reply:
That seems closer. We don't really care about the number of messages sent but we do care about the number of outgoing messages queued, especially when we have unsent messages which the server thinks it already has.